example_copy_test.go

```diff
@@ -215,6 +216,37 @@ func ExampleCopy_remoteToRemote() {
 	// sha256:7cbb44b44e8ede5a89cf193db3f5f2fd019d89697e6b87e8ed2589e60649b0d1
 }
 
+func ExampleCopy_remoteToRemoteWithMount() {
+	reg, err := remote.NewRegistry(remoteHost)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	src, err := reg.Repository(ctx, "source")
+	if err != nil {
+		panic(err) // Handle error
+	}
+	dst, err := reg.Repository(ctx, "target")
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	tagName := "latest"
+
+	opts := oras.CopyOptions{}
+	// Enable cross-repository blob mounting
+	opts.WithMount("source", dst.(registry.Mounter), nil)
```

Comment:
What will happen if someone call `WithMount` twice?

Reply:
This is a fun one.  I think it would work and not produce an error (but I have not tried it).  Let's call the `preCopy` functions `preCopy1`, `preCopy2`, `preCopy3`.  The last one is the one created by the last call to `WithMount`.   The first thing that would happen is the call to `preCopy3` which would attempt a mount.  Let's assume the mount failed, then `getContent` would get called (defined by `preCopy3`) and it would call `preCopy2`.  `preCopy2` was created by the first call to `WithMount` so it will try to mount the blob with that source repo.  Let's assume it also fails, then `getContent` would call `preCopy1`.  Then `getContent` (from `preCopy2`) would return `errdef.ErrUnsupported` (to be renamed) and the blob would be copied.  Then `getContent` (from `preCopy3`) would return `errdef.ErrUnsupported` (to be renamed) and the blob would be copied **again**.  So it seems like the blob would be copied twice with this design.  Not fatal but not ideal either.

Reply:
We could add a field to `CopyOptions`, something like `tryingMount bool`.  In `WithMount()` we can check that `tryingMount` is false (else panic) and then set it to true.  Then if the programmer makes the error to call `WithMount` twice they get a reasonable error.